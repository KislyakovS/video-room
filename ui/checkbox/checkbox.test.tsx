import { render } from '@testing-library/react';
import userEvent from '@testing-library/user-event';

import Checkbox from './checkbox';

interface MockProps {
    type: string;
}

jest.mock('../icon', () => ({ type }: MockProps) => `Svg Icon - ${type}`);

describe('Component Checkbox:', () => {
    it('snapshot', () => {
        const { container } = render(<Checkbox>Checkbox</Checkbox>);

        expect(container).toMatchSnapshot();
    });

    it('should call for a change on the checkbox', async () => {
        const user = userEvent.setup();
        const onChange = jest.fn();

        const { getByRole } = render(<Checkbox onChange={onChange}>Checkbox</Checkbox>);
        const checkbox = getByRole('checkbox') as HTMLInputElement;

        await user.click(checkbox);

        expect(onChange).toHaveBeenCalled();
        expect(checkbox.checked).toBeTruthy();
    });
})