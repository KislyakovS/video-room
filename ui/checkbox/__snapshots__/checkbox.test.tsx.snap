// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Component Checkbox: snapshot 1`] = `
<div>
  <input
    class="input"
    id=":r0:"
    type="checkbox"
  />
  <label
    class="label"
    for=":r0:"
  >
    <span
      class="checked"
    >
      Svg Icon - checked
    </span>
    <span>
      Checkbox
    </span>
  </label>
</div>
`;
